packages/amplify-cli/src/domain/amplify-toolkit.ts

```diff
@@ -461,4 +456,9 @@ export class AmplifyToolkit {
   getCleanUpTasks(): Function[] {
     return this._cleanUpTasks;
   }
+
+  async runCleanUpTasks(context: Context) {
+    const promises = this._cleanUpTasks.map(fn => fn(context));
+    await Promise.all(promises);
```

Comment:
I realize you're structuring the method this way because of how other methods are structured, but why not just do
```suggestion
await Promise.all(this._cleanUpTasks.map(task => task(this)));
```